src/collective/siteimprove/browser/utils.py

```diff
@@ -12,3 +14,16 @@ class AnonCanView(BrowserView):
     def __call__(self):
         """Can we see this"""
         return 'Anonymous' in rolesForPermissionOn('View', self.context)
+class UseDomainView(BrowserView):
+    """Returns true if the domain function is to be used in the siteimprove js header
+       false implies that input function will be used instead"""
+
+    def __call__(self):
+        use_domain = False # use input function by default
+
+        # figure out if domain function should be used instead
+        context_state = getMultiAdapter((self.context, self.request), name=u'plone_context_state')
+        if context_state.is_portal_root():
```

Comment:
I think it would be safer to do the following:

    def __call__(self):
        context_state = queryMultiAdapter((self.context, self.request), name=u'plone_context_state')
        if context_state is not None:
            return context_state.is_portal_root()
        return False